clients/src/main/java/org/apache/kafka/common/record/EosLogEntry.java

```diff
@@ -197,26 +208,29 @@ protected LogRecord makeNext() {
     }
 
     public void setOffset(long offset) {
-        buffer.putLong(OFFSET_OFFSET, offset);
+        buffer.putLong(BASE_OFFSET_OFFSET, offset);
     }
 
+    @Override
     public void setCreateTime(long timestamp) {
         long currentTimestamp = maxTimestamp();
         // We don't need to recompute crc if the timestamp is not updated.
         if (timestampType() == TimestampType.CREATE_TIME && currentTimestamp == timestamp)
             return;
 
         byte attributes = attributes();
-        buffer.put(ATTRIBUTES_OFFSET, TimestampType.CREATE_TIME.updateAttributes(attributes));
-        buffer.putLong(TIMESTAMP_OFFSET, timestamp);
+        buffer.putShort(ATTRIBUTES_OFFSET, TimestampType.CREATE_TIME.updateAttributes(attributes));
+        buffer.putLong(MAX_TIMESTAMP_OFFSET, timestamp);
```

Comment:
In the current usage `timestamp` is guaranteed to be the max timestamp. Perhaps we can change `setCreateTime` and `setLogAppendTime` to make the usage clearer. For example, maybe we should just have a single method: 
```java
void setTimestamp(TimestampType timestampType, long maxTimestamp)
```
To consolidate the two paths, we can leave the BASE_TIMESTAMP field unchanged for both cases. For log append time, we will simply ignore this field. How does that sound?